### PR TITLE
Take a homepage URL into account when enriching VCS info

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/548)

### CODE REVIEW

File: analyzer/src/main/kotlin/PackageManager.kt [hunk 1]
```diff
@@ -130,23 +131,43 @@ abstract class PackageManager {
...
+         * @param homepageUrl The URL to the homepage of a [Package], if any.
          */
-        fun processPackageVcs(vcsFromPackage: VcsInfo): VcsInfo {
+        fun processPackageVcs(vcsFromPackage: VcsInfo, homepageUrl: String = ""): VcsInfo {
             val normalizedVcsFromPackage = vcsFromPackage.normalize()
-            val vcsFromUrl = VersionControlSystem.splitUrl(normalizedVcsFromPackage.url)
+
+            val normalizedUrl = normalizedVcsFromPackage.url.takeIf {
+                it.isNotEmpty()
+            } ?: run {
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/PackageManager.kt]: `run` is not required here.

File: analyzer/src/main/kotlin/managers/Bundler.kt [hunk 2]
```diff
@@ -120,7 +120,7 @@ class Bundler : PackageManager() {
                     declaredLicenses = declaredLicenses.toSortedSet(),
                     aliases = emptyList(),
                     vcs = VcsInfo.EMPTY,
-                    vcsProcessed = processProjectVcs(workingDir),
+                    vcsProcessed = processProjectVcs(projectDir = workingDir, homepageUrl = homepageUrl),
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/Bundler.kt]: Using named parameters here seems a bit verbose.

File: analyzer/src/main/kotlin/managers/PIP.kt [hunk 3]
```diff
@@ -262,7 +265,7 @@ class PIP : PackageManager() {
                 declaredLicenses = declaredLicenses,
                 aliases = emptyList(),
                 vcs = VcsInfo.EMPTY,
-                vcsProcessed = processProjectVcs(workingDir),
+                vcsProcessed = processProjectVcs(projectDir = workingDir, homepageUrl = projectHomepage),
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/PIP.kt]: Using named parameters here seems a bit verbose.

File: analyzer/src/main/kotlin/managers/Bundler.kt (hunk 2 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Bundler.kt]: I did not want to provide a value for the middle `vcsFromProject` parameter, hence the named arguments. Do you prefer passing explicitly passing `VcsInfo.EMPTY` for `vcsFromProject`?

File: analyzer/src/main/kotlin/PackageManager.kt (hunk 1 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/PackageManager.kt]: Done.

File: analyzer/src/main/kotlin/managers/Bundler.kt (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Bundler.kt]: No, I missed that `processProjectVcs` takes three arguments. Actually naming the first parameter would not be required, but I leave this up to you.

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Bundler.kt]: Ok, I've removed the first named arguments.

**Pull request merged**